### ISSUE TITLE
Empty `<a>` tags remaining after link deleted

### ISSUE THREAD

**Issue by jvine** [sul-dlss/exhibits#1761]
There are instances where there is an `<a>` tag remaining in the HTML with no link text. They are in the tab order, but are not visible on the page. Keyboard and SR users will encounter them.

It feels like a curator has highlighted the link in a WYSIWYG and deleted it, but only the text was actually deleted. Note that some include invisible content like `<b>` or `<br>` tags. 

Is there any way to:
a) delete empty links programmatically, and
b) prevent them from happening in the future?

Here's a page with several of them:
https://exhibits.stanford.edu/maps-of-africa/about/resources

Here are some examples from the report:
- `<a href="http://searchworks.stanford.edu/?utf8=â&amp;q=africa&amp;search</i>field=search&amp;f[format][]=Map/Globe"></a>`
- `<a href="http://searchworks.stanford.edu/view/1512347"></a>`
- `<a href="http://searchworks.stanford.edu/view/3871463"></a>`
- `<a href="http://searchworks.stanford.edu/view/5590407"></a>`
- `<a href="http://www-sul.stanford.edu/depts/ssrg/africa/map.html"></a>`
- `<a href="http://www.davidrumsey.com/luna/servlet/view/search?sort=Pub<i>List</i>No<i>InitialSort,Pub</i>Date,Pub<i>List</i>No,Series<i>No&amp;q=africa&amp;search=Go"></a>`
- `<a href="http://www.lib.utexas.edu/maps/historical/history</i>africa.html"></a>`
- `<a href="http://www.lib.utexas.edu/maps/historical/history</i>africa.html"><br></a>`
- `<a href="http://www.library.illinois.edu/contentdm/cdm4/browse.php?CISOROOT=/africanmaps"></a>`
- `<a href="http://www.oac.cdlib.org/findaid/ark:/13030/kt2k4035s1/admin/?query=knuth#hitNum10"></a>`
- `<a href="http://www.oac.cdlib.org/findaid/ark:/13030/kt2k4035s1/admin/?query=knuth#hitNum11"></a>`
- `<a href="http://www.oac.cdlib.org/findaid/ark:/13030/kt2k4035s1/admin/?query=knuth#hitNum12"></a>`
- `<a href="http://www.oac.cdlib.org/findaid/ark:/13030/kt2k4035s1/admin/?query=knuth#hitNum9"></a>`
- `<a href="http://www.oac.cdlib.org/findaid/ark:/13030/kt787007h3/"></a>`
- `<a href="http://www.press.uchicago.edu/Misc/Chicago/907287.html"></a>`
- `<a href="https://en.wikipedia.org/wiki/Space_Shuttle<i>Endeavour"></a>`
- `<a href="https://en.wikipedia.org/wiki/Space_Shuttle<i>Endeavour"></a>`
- `<a href="https://exhibits.stanford.edu/data/catalog/hs422jm3330"></a>`
- `<a href="https://exhibits.stanford.edu/mjf/feature/historic-performances"></a>`
- `<a href="https://exhibits.stanford.edu/mjf/feature/the-festival-first-hand-oral-histories-and-interviews"></a>`
- `<a href="https://exhibits.stanford.edu/mjf/feature/the-festival-first-hand-oral-histories-and-interviews"></a>`
- `<a href="https://exhibits.stanford.edu/mjf/feature/the-festival-first-hand-oral-histories-and-interviews"><br></a>`
- `<a href="https://exhibits.stanford.edu/sftjf/catalog/vr427hg6850"><b></b></a>`
- `<a href="https://exhibits.stanford.edu/sftjf/catalog/vr427hg6850"><b></b></a>`
- `<a href="https://exhibits.stanford.edu/sftjf/feature/bill-dart"></a>`
- `<a href="https://exhibits.stanford.edu/sftjf/feature/bob-helm"></a>`
- `<a href="https://exhibits.stanford.edu/sftjf/feature/wally-rose"></a>`
- `<a href="https://library.stanford.edu/blogs/stanford-libraries-blog/2015/06/playasax-acquired-ars"></a>`
- `<a href="https://library.stanford.edu/blogs/stanford-libraries-blog/2018/12/player-piano-roll-repair-report-conservation"></a>`
- `<a href="https://library.stanford.edu/blogs/stanford-libraries-blog/2018/12/player-piano-roll-repair-report-conservation"></a>`
- `<a href="https://library.stanford.edu/blogs/stanford-libraries-blog/2018/12/player-piano-roll-repair-report-conservation"></a>`
- `<a href="https://library.stanford.edu/blogs/stanford-libraries-blog/2018/12/player-piano-roll-repair-report-conservation"></a>`
- `<a href="https://library.stanford.edu/blogs/stanford-libraries-blog/2018/12/player-piano-roll-repair-report-conservation"></a>`
- `<a href="https://library.stanford.edu/blogs/stanford-libraries-blog/2018/12/player-piano-roll-repair-report-conservation"></a>`
- `<a href="https://news.stanford.edu/2019/05/03/leonardo-da-vinci-celebrated-stanfords-green-library/"><br></a>`
- `<a href="https://searchworks.stanford.edu/view/4085977"><br></a>`
- `<a href="https://searchworks.stanford.edu/view/4719997"></a>`
- `<a href="https://searchworks.stanford.edu/view/4719997"></a>`
- `<a href="https://searchworks.stanford.edu/view/4719997"></a>`
- `<a href="https://web.stanford.edu/dept/SUL/library/sunset-magazine/html/1930_may.html"></a>`
- `<a href="https://web.stanford.edu/dept/SUL/library/sunset-magazine/html/1977_may.html"></a>`
- `<a href="https://web.stanford.edu/dept/SUL/library/sunset-magazine/html/writers.html"></a>`
- `<a href="https://www.myscience.org/news/wire/leonardo_da_vinci_is_celebrated_at_stanford_s_green_library-2019-stanford"></a>`
- `<a href="https://www.paloaltoonline.com/news/2019/05/29/whats-in-leonardos-library"></a>`
- `<a href="mailto:specialcollections@stanford.edu"></a>`

**Comment by mejackreed** [comment]
I have tried a few different ways to recreate this, and haven't been able to using the WYSIWYG editor. @jvine is this rampant in other exhibits or just this one?

**Comment by jvine** [comment]
There are lots in Maps of Africa, but also in
- Leonardo's Library
- Stanford University Piano Roll Archive
- The Baltic Video Archive
- Sanborn Fire Insurance Maps
- Stanford Stories From the Archives

.... I'm trying to run a complete report, but it will take a little while.

**Comment by mejackreed** [comment]
No problem.. this helps a lot and its ok without the report.

**Comment by mejackreed** [comment]
Maybe useful for those looking into this:

```ruby
Spotlight::Page.all.map { |p| p.content.map { |c| c.text =~ /<a href="https?:\/\/[\S]+"><\/a>/ }}.flatten.compact
```

**Comment by jvine** [comment]
Adding:

- Copyright Renewals
- Sunset Magazine
- The San Francisco Traditional Jazz Foundation Collection
- Research from Stanford University
- Transcontinental Railroad
- Women @ Stanford

and there are almost certainly more but the report maxed out.
